File: server_src/main/java/sdu/edu/pojo/APK.java
package sdu.edu.pojo;

import java.util.Date;

public class APK {
    private String MD5;
    private String name;
    private double size;
    private double RiskLevel;
    private double API_score;
    private double permission_score;
    private double behavior_scores;
    private String[] permission;
    private String[] Risk_Function;
    private String[] Activities;
    private String[] TelephonyIdentifiersLeakage;
    private String[] DeviceSettingsHarvesting;
    private String[] LocationLookup;
    private String[] ConnectionInterfacesExfiltration;
    private String[] TelephonyServicesAbuse;
    private String[] PimDataLeakage;
    private String[] SuspiciousConnectionEstablishment;
    private String[] AudioVideoEavesdropping;
    private String[] CodeExecution;
    private String UploadDate;
    private String status;
    private String packagename;
    private String MinSdkVersion;


    public double getBehavior_scores() {
        return behavior_scores;
    }

    public void setBehavior_scores(double behavior_scores) {
        this.behavior_scores = behavior_scores;
    }

    public String[] getCodeExecution() {
        return CodeExecution;
    }

    public void setCodeExecution(String[] codeExecution) {
        CodeExecution = codeExecution;
    }


    public String[] getTelephonyIdentifiersLeakage() {
        return TelephonyIdentifiersLeakage;
    }

    public void setTelephonyIdentifiersLeakage(String[] telephonyIdentifiersLeakage) {
        TelephonyIdentifiersLeakage = telephonyIdentifiersLeakage;
    }

    public String[] getDeviceSettingsHarvesting() {
        return DeviceSettingsHarvesting;
    }

    public void setDeviceSettingsHarvesting(String[] deviceSettingsHarvesting) {
        DeviceSettingsHarvesting = deviceSettingsHarvesting;
    }

    public String[] getLocationLookup() {
        return LocationLookup;
    }

    public void setLocationLookup(String[] locationLookup) {
        LocationLookup = locationLookup;
    }

    public String[] getConnectionInterfacesExfiltration() {
        return ConnectionInterfacesExfiltration;
    }

    public void setConnectionInterfacesExfiltration(String[] connectionInterfacesExfiltration) {
        ConnectionInterfacesExfiltration = connectionInterfacesExfiltration;
    }

    public String[] getTelephonyServicesAbuse() {
        return TelephonyServicesAbuse;
    }

    public void setTelephonyServicesAbuse(String[] telephonyServicesAbuse) {
        TelephonyServicesAbuse = telephonyServicesAbuse;
    }

    public String[] getPimDataLeakage() {
        return PimDataLeakage;
    }

    public void setPimDataLeakage(String[] pimDataLeakage) {
        PimDataLeakage = pimDataLeakage;
    }

    public String[] getSuspiciousConnectionEstablishment() {
        return SuspiciousConnectionEstablishment;
    }

    public void setSuspiciousConnectionEstablishment(String[] suspiciousConnectionEstablishment) {
        SuspiciousConnectionEstablishment = suspiciousConnectionEstablishment;
    }

    public String[] getAudioVideoEavesdropping() {
        return AudioVideoEavesdropping;
    }

    public void setAudioVideoEavesdropping(String[] audioVideoEavesdropping) {
        AudioVideoEavesdropping = audioVideoEavesdropping;
    }

    public String getMinSdkVersion() {
        return MinSdkVersion;
    }

    public double getAPI_score() {
        return API_score;
    }

    public void setAPI_score(double API_score) {
        this.API_score = API_score;
    }

    public double getPermission_score() {
        return permission_score;
    }

    public void setPermission_score(double permission_score) {
        this.permission_score = permission_score;
    }

    public void setMinSdkVersion(String minSdkVersion) {
        MinSdkVersion = minSdkVersion;
    }


    public String getMD5() {
        return MD5;
    }

    public void setMD5(String MD5) {
        this.MD5 = MD5;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public double getSize() {
        return size;
    }

    public void setSize(double size) {
        this.size = size;
    }

    public double getRiskLevel() {
        return RiskLevel;
    }

    public void setRiskLevel(double riskLevel) {
        RiskLevel = riskLevel;
    }

    public String[] getPermission() {
        return permission;
    }

    public void setPermission(String[] permission) {
        this.permission = permission;
    }

    public String[] getRisk_Function() {
        return Risk_Function;
    }

    public void setRisk_Function(String[] risk_Function) {
        Risk_Function = risk_Function;
    }

    public String[] getActivities() {
        return Activities;
    }

    public void setActivities(String[] activities) {
        Activities = activities;
    }

    public String getUploadDate() {
        return UploadDate;
    }

    public void setUploadDate(String uploadDate) {
        UploadDate = uploadDate;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getPackagename() {
        return packagename;
    }

    public void setPackagename(String packagename) {
        this.packagename = packagename;
    }
}

File: ml_src/APK.py
class APK:
    def __init__(self, RiskFunction=dict(),behavior_scores=0,MinSdkVersion='', Permission=list(), Activities=list(), API_score=0,
                 permission_score=0, PackageName='', Status='', TelephonyIdentifiersLeakage=list(),
                 DeviceSettingsHarvesting=list(), LocationLookup=list(), ConnectionInterfacesExfiltration=list(),
                 TelephonyServicesAbuse=list(), PimDataLeakage=list(), SuspiciousConnectionEstablishment=list(),
                 AudioVideoEavesdropping=list(), CodeExecution=list()):
        self.RiskFunction = RiskFunction
        self.Permission = Permission
        # 修改了这里
        self.API_score = API_score
        self.permission_score = permission_score
        self.RiskLevel = lambda x,y: (x+y)/2
        self.PackageName = PackageName
        self.Status = Status
        self.Activities = Activities
        self.MinSdkVersion = MinSdkVersion
        self.TelephonyIdentifiersLeakage = TelephonyIdentifiersLeakage
        self.DeviceSettingsHarvesting = DeviceSettingsHarvesting
        self.LocationLookup = LocationLookup
        self.ConnectionInterfacesExfiltration = ConnectionInterfacesExfiltration
        self.TelephonyServicesAbuse = TelephonyServicesAbuse
        self.PimDataLeakage = PimDataLeakage
        self.SuspiciousConnectionEstablishment = SuspiciousConnectionEstablishment
        self.AudioVideoEavesdropping = AudioVideoEavesdropping
        self.CodeExecution = CodeExecution
        self.behavior_scores = behavior_scores

    def todict(self):
        data = dict()
        data['behavior_scores'] = self.behavior_scores
        data['RiskFunction'] = self.RiskFunction
        data['Permission'] = self.Permission
        data['RiskLevel'] = int(self.RiskLevel(self.API_score,self.permission_score))
        data['API_score'] = self.API_score
        data['permission_score'] = self.permission_score
        data['PackageName'] = self.PackageName
        data['Status'] = self.Status
        data['Activities'] = self.Activities
        data['MinSdkVersion'] = self.MinSdkVersion
        data['TelephonyIdentifiersLeakage'] = self.TelephonyIdentifiersLeakage
        data['DeviceSettingsHarvesting'] = self.DeviceSettingsHarvesting
        data['LocationLookup'] = self.LocationLookup
        data['ConnectionInterfacesExfiltration'] = self.ConnectionInterfacesExfiltration
        data['TelephonyServicesAbuse'] = self.TelephonyServicesAbuse
        data['PimDataLeakage'] = self.PimDataLeakage
        data['SuspiciousConnectionEstablishment'] = self.SuspiciousConnectionEstablishment
        data['AudioVideoEavesdropping'] = self.AudioVideoEavesdropping
        data['CodeExecution'] = self.CodeExecution
        return data
